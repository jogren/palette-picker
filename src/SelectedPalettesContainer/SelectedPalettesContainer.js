import React from 'react';
import { connect } from 'react-redux';
import deleteIcon from '../images/delete.png';
import { FiTrash2 } from 'react-icons/fi';

export const SelectedPalettesContainer = ({ selectedPalettes, deletePalette, editPalette }) => {

  let selectedPalettesDisplay = selectedPalettes.map(palette => {
      return <div className="selected-palette" key={palette.name}>
        <header>
          <p>{palette.name}</p>
          <FiTrash2 className="trash-img" onClick={() => deletePalette(palette.id, palette.project_id)} src={deleteIcon} alt={palette.name}/>
        </header>
        <section className="color-container" onClick={() => editPalette(palette.id)}>
          <div style={{ backgroundColor: palette.color1 }}></div>
          <div style={{ backgroundColor: palette.color2 }}></div>
          <div style={{ backgroundColor: palette.color3 }}></div>
          <div style={{ backgroundColor: palette.color4 }}></div>
          <div style={{ backgroundColor: palette.color5 }}></div>
        </section>
      </div>
  })
  
  return (
    <section className="selectedPalettes_section">
      { selectedPalettesDisplay }
      {!selectedPalettes.length && <p className="no-selected-palettes">Please select a project or add some palettes</p> }
    </section>
  )
}

export const mapStateToProps = ({ selectedPalettes }) => ({
  selectedPalettes
});

export default connect(mapStateToProps)(SelectedPalettesContainer);