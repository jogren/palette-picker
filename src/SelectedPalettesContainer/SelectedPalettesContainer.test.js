import React from 'react';
import { shallow } from 'enzyme';
import { SelectedPalettesContainer, mapStateToProps } from './SelectedPalettesContainer';

describe('SelectedPalettesContainer', () => {
  let wrapper;
  const deletePaletteMock = jest.fn();
  const editPaletteMock = jest.fn();
  const selectedPalettesMock = [{
    id: 21,
    name: 'palette',
    project_id: 7,
    created_at: '2019-10-12T23:36:53.454Z',
    updated_at: '2019-10-12T23:36:53.454Z',
    color1: '#5fd849',
    color2: '#878dff',
    color3: '#efa58b',
    color4: '#f2bf59',
    color5: '#f73722'
  }];
  const currentPaletteMock = [{
    hexCode: "#25b25d",
    isLocked: false
  }];
  const currentProjectsMock = [{
    id: 13,
    name: '123456',
    created_at: '2019-10-13T17:40:38.824Z',
    updated_at: '2019-10-13T17:40:38.824Z'
  }];
  const currentPaletteIdMock = {
    id: 49,
    name: '12345',
    projectId: 13
  };
  const mockState = {
    currentPalette: currentPaletteMock,
    currentProjects: currentProjectsMock,
    selectedPalettes: selectedPalettesMock,
    currentPaletteId: currentPaletteIdMock,
    errorMsg: ""
  }

  beforeEach(() => {
    wrapper = shallow(<SelectedPalettesContainer
      deletePalette={deletePaletteMock}
      selectedPalettes={selectedPalettesMock}
      editPalette={editPaletteMock}
      />)
  });

  it('should match the snapshot with the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('mapStateToProps should grab the props it needs', () => {
    const expected = {
      selectedPalettes: selectedPalettesMock
    }

    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected);
  });

  it('should call the editPalette method when the palette is clicked', () => {
    wrapper.find('section').at(1).simulate('click');
    expect(editPaletteMock).toHaveBeenCalled();
  });

  it('should call the deletePalette method when the trash image is clicked', () => {
    wrapper.find('FiTrash2').simulate('click');
    expect(deletePaletteMock).toHaveBeenCalled();
  });
});