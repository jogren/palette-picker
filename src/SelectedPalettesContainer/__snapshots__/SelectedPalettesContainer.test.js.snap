// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedPalettesContainer should match the snapshot with the data passed through 1`] = `
<section
  className="selectedPalettes_section"
>
  <div
    className="selected-palette"
    key="palette"
  >
    <header>
      <p>
        palette
      </p>
      <img
        alt="palette"
        className="img_color-container"
        onClick={[Function]}
        src="delete.png"
      />
    </header>
    <section
      className="color-container"
      onClick={[Function]}
    >
      <div
        style={
          Object {
            "backgroundColor": "#5fd849",
          }
        }
      />
      <div
        style={
          Object {
            "backgroundColor": "#878dff",
          }
        }
      />
      <div
        style={
          Object {
            "backgroundColor": "#efa58b",
          }
        }
      />
      <div
        style={
          Object {
            "backgroundColor": "#f2bf59",
          }
        }
      />
      <div
        style={
          Object {
            "backgroundColor": "#f73722",
          }
        }
      />
    </section>
  </div>
</section>
`;
